client/src/modules/depots/modals/depot.modal.js

```diff
@@ -0,0 +1,42 @@
+angular.module('bhima.controllers')
+  .controller('DepotModalController', DepotModalController);
+
+DepotModalController.$inject = [
+  '$state', 'ServiceService', 'DepotService',
+  'SessionService', 'ModalService', 'NotifyService',
+];
+
+function DepotModalController($state, Services, Depots,
+  SessionService, ModalService, Notify) {
+  var vm = this;
+
+  vm.depot = $state.params.depot;
+  vm.isCreating = !!($state.params.creating);
+
+  // exposed methods
+  vm.submit = submit;
+  vm.closeModal = closeModal;
+
+  // submit the data to the server from all two forms (update, create)
+  function submit(depotForm) {
+    var promise;
+
+    if (depotForm.$invalid || !depotForm.$dirty) { return 0; }
+
+    promise = (vm.isCreating) ?
+      Depots.create(vm.depot) :
+      Depots.update(vm.depot.uuid, vm.depot);
+
+    return promise
+      .then(function () {
+        var translateKey = (vm.isCreating) ? 'DEPOT.CREATED' : 'DEPOT.UPDATED';
+        Notify.success(translateKey);
+        $state.go('depots', null, { reload : true });
+      })
+      .catch(Notify.handleError);
+  }
+
+  function closeModal() {
+    $state.transitionTo('depots');
```

Comment:
It is much better to use `$state.go()` that `$state.transitionTo()`.  The first one uses the second one internally.  To ensure we have the same behavior across all our modules, let us continue to use `$state.go()`.